Mathlib/CategoryTheory/Monoidal/Braided/Reflection.lean

```diff
@@ -0,0 +1,233 @@
+      erw [Adjunction.homEquiv_unit, Adjunction.homEquiv_counit]
+      simp
+    rw [w₃, isIso_iff_bijective]
+    simp only [comp_obj, op_tensorObj, coyoneda_obj_obj, unop_tensorObj, id_obj,
+      yoneda_obj_obj, tensorLeft_obj, EquivLike.comp_bijective, EquivLike.bijective_comp]
+    have w₄ : (coyoneda.map (adj.unit.app d).op).app ((ihom d').obj (R.obj c)) ≫
+        (coyoneda.obj ⟨d⟩).map (adj.unit.app ((ihom d').obj (R.obj c))) =
+          (coyoneda.obj ⟨(L ⋙ R).obj d⟩).map (adj.unit.app ((ihom d').obj (R.obj c))) ≫
+            (coyoneda.map (adj.unit.app d).op).app _ := by simp
+    rw [← isIso_iff_bijective]
+    suffices IsIso ((coyoneda.map (adj.unit.app d).op).app ((ihom d').obj (R.obj c)) ≫
+        (coyoneda.obj ⟨d⟩).map (adj.unit.app ((ihom d').obj (R.obj c)))) from
+      IsIso.of_isIso_comp_right _ ((coyoneda.obj ⟨d⟩).map (adj.unit.app ((ihom d').obj (R.obj c))))
+    rw [w₄]
+    refine IsIso.comp_isIso' inferInstance ?_
+    constructor
+    -- We give the inverse of the bottom map in the stack of commutative squares:
+    refine ⟨fun f ↦ R.map ((adj.homEquiv _ _).symm f), ?_, by ext; simp⟩
+    ext f
+    simp only [comp_obj, coyoneda_obj_obj, id_obj, Adjunction.homEquiv_counit,
+      map_comp, types_comp_apply, coyoneda_map_app, Quiver.Hom.unop_op, assoc,
+      types_id_apply]
+    have : f = R.map (R.preimage f) := by simp
+    rw [this]
+    simp [← map_comp, ← map_comp_assoc, -map_preimage]
+  tfae_have 2 ↔ 3
+  · conv => lhs; intro c d; rw [isIso_iff_isIso_yoneda_map]
+    conv => rhs; intro d d'; rw [isIso_iff_isIso_coyoneda_map]
+    -- bring the quantifiers out of the `↔`:
+    rw [forall_swap]; apply forall_congr'; intro d
+    rw [forall_swap]; apply forall₂_congr; intro d' c
+    -- `w₁, w₂,` are the two stacked commutative squares in the proof on nLab:
+    have w₁ : ((coyoneda.map (L.map (adj.unit.app d ▷ d')).op).app c) =
+        (adj.homEquiv _ _).symm ∘
+          (coyoneda.map (adj.unit.app d ▷ d').op).app (R.obj c) ∘
+            (adj.homEquiv _ _) := by ext; simp
+    have w₂ : ((yoneda.map ((pre (adj.unit.app d)).app (R.obj c))).app ⟨d'⟩) =
+          ((ihom.adjunction d).homEquiv _ _) ∘
+            ((coyoneda.map (adj.unit.app d ▷ d').op).app (R.obj c)) ∘
+              ((ihom.adjunction ((L ⋙ R).obj d)).homEquiv _ _).symm := by
+      rw [← Function.comp.assoc, ((ihom.adjunction ((L ⋙ R).obj d)).homEquiv _ _).eq_comp_symm]
+      ext
+      simp only [id_obj, yoneda_obj_obj, comp_obj, Function.comp_apply,
+        yoneda_map_app, op_tensorObj, coyoneda_obj_obj, unop_tensorObj, op_whiskerRight,
+        coyoneda_map_app, unop_whiskerRight, Quiver.Hom.unop_op]
+      erw [Adjunction.homEquiv_unit, Adjunction.homEquiv_unit]
+      simp
+    rw [w₂, w₁, isIso_iff_bijective, isIso_iff_bijective]
+    simp
+  tfae_finish
+
+end
+
+section
+variable [MonoidalCategory C]
+variable {L : MonoidalFunctor D C} {R : C ⥤ D} [R.Faithful] [R.Full] (adj : L.toFunctor ⊣ R)
+
+include adj in
```

Comment:
Do you need this? After all `adj` is explicitly mentioned in the statement. Idem below.